src/main/java/uk/gov/justice/probation/courtcasematcher/messaging/MessageProcessor.java

```diff
@@ -34,8 +34,9 @@ default void logErrors(Logger log, CourtCaseFailureEvent courtCaseFailureEvent)
 
     default String saveCase(Case aCase, String messageId) {
         getTelemetryService().trackCourtCaseEvent(aCase, messageId);
-        getCourtCaseService().getCourtCase(aCase)
-            .subscribe(this::postCaseEvent);
+        final var courtCase = getCourtCaseService().getCourtCase(aCase)
+                .block();
+        postCaseEvent(courtCase);
         return aCase.getCaseNo();
```

Comment:
There's no use of the return value from this method. Wonder if we drop it, or maybe log the case ID instead of case no in the caller.